packages/flatbuffers/package.py

```diff
@@ -26,7 +26,7 @@ class Flatbuffers(CMakePackage):
     def cmake_args(self):
         args=[]
         args.append('-DFLATBUFFERS_BUILD_SHAREDLIB=ON')
-        args.append('-DFLATBUFFERS_BUILD_FLATLIB=OFF')
+        args.append('-DFLATBUFFERS_BUILD_FLATLIB=ON')
```

Comment:
Do you actually need static libraries?
They break fairmq often.

Reply:
ok, reverted back. But we need to make change in FairRoot that it does not search for static library of flatbuffers.

Reply:
Okay, that's a little strange.

@dennisklein can you please comment on the following, I seem to miss something.

* flatbuffers can be built as static and/or shared libraries.
* If building static libraries at all, fairmq (some versions?) tried to link a static library into a shared object. This fails. Why fairmq isn't using the shared flatbuffers: I don't know.
* If building only shared flatbuffers, fairroot seems to have an issue: Why can't it link to the shared library?

What should we do?

Reply:
The Flatbuffers library is exports two different names for the static and shared library: `flatbuffers::flatbuffers` and `flatbuffers::flatbuffers_shared`. FairMQ initially only used `flatbuffers::flatbuffers` and only some patch release versions later gained the CMake logic to use `flatbuffers::flatbuffers_shared` is the other is not available.

FairRoot currently only relies on the `flatbuffers::flatbuffers` name.

Reply:
The only requirement is that the static flatbuffers is built with PIC. For versions that do not use the shared name, the relevant lines to patch are:

https://github.com/FairRootGroup/FairRoot/blob/4ab2d2b1f0392ad93b92026d70fe049c38665f7d/examples/advanced/Tutorial3/CMakeLists.txt#L127

Reply:
Hm, FairMQ should always support both names, the Flatbuffers dependency was introduced in `v1.4.9`.

Reply:
Shouldn't the logic be the other way round?
First check for `flatbuffers::flatbuffers_shared`, and if that does not exist, use `flatbuffers::flatbuffers`? When creating a shared library/object (fairmq! fairoot?), one need the library as a shared library.

Reply:
Sure, I don't mind switching the logic, if you prefer. But why is it needed?

Btw, usually, most CMake packages export only one name and either populate it with the static or the shared library.

Reply:
Okay, after some private discussions, the current conclusion seems to be:

* Let fairroot use the shared version of flatbuffers. @dennisklein will take care of this.
* As a quick short fix, one could remove the `depends_on('flatbuffers')` in `fiarroot/package.py`.

Reply:
FairRootGroup/FairRoot#936